File: src/core/notifications/notification.js
import { ref } from "vue";
import NOTIFICATION_TYPE from "./notificationType";

export class Notification {
	constructor() {
		this.type = ref(NOTIFICATION_TYPE.None);
		this.title = ref("");
		this.body = ref("");
		this.code = ref("");
		this.duration = ref(0);
	}

	from_json(json) {
		this.type = ref(json.type);
		this.title = ref(json.title);
		this.body = ref(json.body);
		this.code = ref(json.code);
		this.duration = ref(json.duration);
	}
}

File: src-tauri/src/notifications/notification.rs
use serde::Serialize;
use tauri::Manager;

use crate::MAIN_APP;

#[derive(Clone, Serialize)]
pub enum NotificationType {
    None,
    Info,
    Warning,
    Error,
}

#[derive(Clone, Serialize)]
pub struct Notification {
    pub r#type: NotificationType,
    pub title: String,
    pub body: String,
    pub duration: u32,
    pub code: String,
}

impl Notification {
    pub async fn send(self) {
        let app = MAIN_APP.lock().await;
        if let Some(app) = &*app {
            match app.emit("notification", self) {
                Ok(_) => (),
                Err(_) => println!("[bot::category_create()] Could not emit to windows"),
            }
        } else {
            println!("[bot::category_create()] No app found");
        }
    }
}
